### DISCUSSION TITLE
how to send the invitation email

### DISCUSSION THREAD

**Discussion by VinothGox** [chaskiq/chaskiq#632] [Q&A] [answered]
Hi there,  

Recently I saw your chaskiq tool really is very nice. 

chaskiq app deployed the Nginx server. All setup finished and working is fine in development mood. But invitation mail is not delivered. 
all env variable is correct but I am unable to send the invitation mail.

Error message: "Skipped delivery of mail 6202b4d9a6ca8_44d9a9884888b@e2e-88-7.mail as `perform_deliveries` is false"

after I set the true "perform_deliveries: true" in development.rb file  I get another error (Internal Server Error)
the error is "unable to sign request without credentials set" 

Please help me, past two days I facing this problem. I can't resolve it.

Regards,
Vinoth

**Comment by michelson** [comment, marked as answer]
Hi, thanks for the detailed explanation.
If you perform deliveries in development mode, you will need to set up your SMTP credentials.

the following are some environment variables to set up your email service:

```
SMTP_ADDRESS=
SMTP_PASSWORD=
SMTP_USERNAME=

SMTP_DELIVERY_METHOD=ses
```
you need to put those env variables in the .env file in the project. *Note that in a production environment, the .env file and the methods to add env vars may vary from cloud to cloud.*

`ses` method is relevant for some specific behavior in order to receive messages from emails threads. Of course, if you don't use ses, then you entirely lose that feature, but the other email dispatching will work just fine.

**Reply by michelson** [reply]
note that in development env, the email delivery is not needed (so it is better to set perform_deliveries = false); you can see directly into the app logs on the rails console to see the email content that was sent ;)

**Reply by VinothGox** [reply]
Wow, First of all very thanks for the quick reply.  I forget to add SMTP details in the env file.

Note: your env configuration document is not updated. that document does not specify the SMTP details.
https://dev.chaskiq.io/production-configuration/enviroment-variables-configuration

**Reply by VinothGox** [reply]
sorry, One more question your env variable only working in development mood. am I right? because I try to run the production mood but I am unable to do this.

https://github.com/chaskiq/chaskiq/issues/570#issuecomment-998821353

**Reply by michelson** [reply]
yeah, SMTP credentials are not really needed, if you use amazon SES the aws credentials will just work. 
.env file will not work in production, you will need to set your ENV vars on the cloud provider or your VPS

**Reply by VinothGox** [reply]
Thanks for your response.

**Reply by VinothGox** [reply]
Thanks, Michelson now the invitation is working fine :)

**Comment by f10atin9** [comment]
I think maybe it would be better to support configuring `perform_deliveries` using environment variables?

**Reply by michelson** [reply]
the development environment is not suited for perform_deliveries.

**Reply by f10atin9** [reply]
> the development environment is not suited for perform_deliveries.

How should the official environment be set up?

**Reply by michelson** [reply]
in the production environment, the perform_deliveres option is activated. Here is an article in regard of environment variables

https://dev.chaskiq.io/production-configuration/enviroment-variables-configuration

**Reply by f10atin9** [reply]
> in the production environment, the perform_deliveres option is activated. Here is an article in regard of environment variables
> 
> https://dev.chaskiq.io/production-configuration/enviroment-variables-configuration

Thank you very much, I didn't notice these variables when I read the installation documentation before
